blocks/bricks/recurrent.py

```diff
@@ -349,8 +348,7 @@ def state_to_reset(self):
     def get_dim(self, name):
         if name == 'mask':
             return 0
-        if name in (GatedRecurrent.apply.sequences
-                    + GatedRecurrent.apply.states):
+        if name in (self.apply.sequences + self.apply.states):
```

Comment:
But OK, having polymorphism here kind of make sense, unlike the case with the `AttentionTransition`. There a delegate application is specified and there should be nothing polymorphic about that.